src/actions/targetActions.js

```diff
@@ -45,3 +61,24 @@ export const addTarget = target =>
     }
   };
 
+export const selectTarget = target =>
+  (dispatch) => {
+    target.id ? 
+      (
+        dispatch(selectTargetSuccess(target))
+      ) :
+      (
+        dispatch(unSelectTargetSuccess(target))
+      );
+  };
+
+export const removeTarget = (target, index) =>
+  async (dispatch) => {
+    try {
+      const targetResponse = await TargetApi.deleteTarget(target);
+      dispatch(deleteTargetSuccess(targetResponse, index));
+    } catch (err) {
+      dispatch(deleteTargetSuccess({}, index));
```

Comment:
You should not use a function called `deleteTargetSuccess` when it fails. That's supposed to be used on the successful case.

Reply:
The method 'Delete' for some reason always is rejected